File: main_raspy/communication.py
# ========================================================================
#                 🔹 C H E S S B O T  -  Z E R O 🔹
# ========================================================================
#   Archivo    : communication.py
#   Autor      : Klaus Michalsky
#   Fecha      : Feb. 2026
# ------------------------------------------------------------------------
#   ▫️ DESCRIPCIÓN
#       - Comunicación UART con el RP2040 Zero
#   ▫️ RESPONSABILIDADES
#       - Inicializar UART
#       - Enviar y recibir datos
# ========================================================================


from machine import UART
import config

# CREACIÓN DE UART
# ------------------------------------------------------------------------
uart = UART(
    config.UART_ID,
    config.UART_BAUD,
    tx=config.UART_TX_PIN,
    rx=config.UART_RX_PIN
)


def write(data):
    uart.write(data)


def any():
    return uart.any()


def readline():
    return uart.readline()

File: communication.cpp
// =======================================================================
//                 🔹 C H E S S B O T  —   Z E R O 🔹
// =======================================================================
//  Archivo    : communication.cpp
//  Autor      : Klaus Michalsky
//  Fecha      : Feb-2026
// -----------------------------------------------------------------------
//  ▫️ DESCRIPCIÓN
//      - Inicializar UART
//      - Comunicación UART con el Raspberry
//      - Gestionar la recepción de datos de forma no bloqueante.
// =======================================================================

#include <Arduino.h>

#include "communication.h"
#include "config.h"

// CONFIGURACIÓN DE DEPURACIÓN UART
// -----------------------------------------------------------------------
void debug(const String &msg) {
#if DEBUG_UART
    Serial1.println("Comando recibido: " + msg);
#endif
}

// INICIALIZACIÓN DE COMUNICACION UART
// -----------------------------------------------------------------------
void communicationInit() {
    // USB para debug (opcional)
    Serial.begin(115200);

    // UART hardware en pines 0 y 1
    Serial1.setTX(0);
    Serial1.setRX(1);
    Serial1.begin(115200);
    while (Serial1.available())
        Serial1.read(); // limpia buffer UART
}
